src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -438,6 +450,8 @@ DistributedSearchTreeImpl<DeviceType>::queryDispatchImpl(
       filterResults(space, queries, distances, indices, offset, ranks);
     }
```

Comment:
This loop has to have finer profiling region granularities.

Reply:
I think there is not the one right solution. Do we really want to expose the slightest details here?

Reply:
Not the slightest. But at the minimum, we need to separate top and bottom tree searches and postprocessing.